sql/core/src/test/scala/org/apache/spark/sql/streaming/test/DataFrameReaderWriterSuite.scala

```diff
@@ -572,4 +572,25 @@ class DataFrameReaderWriterSuite extends StreamTest with BeforeAndAfter {
 
     cq.awaitTermination(2000L)
   }
+
+  test("foreach") {
+    import testImplicits._
+
+    val ds = spark.read
+      .format("org.apache.spark.sql.streaming.test")
+      .stream()
+      .as[Int]
+
+    val cq = ds.write
+      .format("console")
+      .option("checkpointLocation", newMetadataDir)
+      .trigger(ProcessingTime(2.seconds))
+      .foreach(new ForeachWriter[Int] {
+        override def open(partitionId: Long, version: Long): Boolean = true
+        override def process(value: Int): Unit = {}
+        override def close(errorOrNull: Throwable): Unit = {}
+      })
+
+    cq.awaitTermination(2000L)
```

Comment:
what does this test? whether it started correctly? Should it not set a global variable to make sure that this actually works rather than implicitly depending on behavior of awaitTermination?